### DISCUSSION TITLE
slow download speed for youtube videos

### DISCUSSION THREAD

**Discussion by lumpsoid** [streamlink/streamlink#5171] [Q&A] [answered]
### Description

- [ ] This post is not about a plugin issue, a bug report, a plugin request or a feature request 
- [x] [I have read the contribution guidelines] https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [x] I have validated that I'm using the [latest version of Streamlink](https://streamlink.github.io/changelog.html) via the [--version](https://streamlink.github.io/cli.html#cmdoption-version) or [--loglevel=debug](https://streamlink.github.io/cli.html#cmdoption-loglevel) parameter
- [x]  [I have read the Streamlink documentation and tried looking for answers to my questions](https://streamlink.github.io/)

```
streamlink 5.2.1
linux 6.1.11.arch1-1
```

YouTube videos are downloading at 50 Kbytes/sec when my internet is ~25000 Kbytes/sec, which is very strange because it was fine two to three months ago when I first started using streamlink.

Checked how that compares to streamlink download speeds from Twitch, and there's more or less -- ~1490Kbytes/sec.

Tried searching for something in issues, but couldn't find anything relevant.

I only found something about using threads, but it didn't change anything.

I don't seem to have installed any programs that interferes with my internet connection and haven't touched that aspect of my computer much.

I have tried both uploading videos to a file and streaming, the main problem is with streaming, but the uploading behaves the same.

screenshot of twitch and youtube download tests
![image](https://user-images.githubusercontent.com/79290581/219367286-1cb175ad-4692-442b-a98e-221c22e23ff8.png)

screenshot of the threading test
![image](https://user-images.githubusercontent.com/79290581/219369244-a9c09ea6-5dcd-4994-9796-79ce632b1dd2.png)

[pastebin loglevel=trace](https://pastebin.com/1uNEscrX)

tried reinstalling `streamlink`, `youtube-dl`, `mpv`
tried replacing `youtube-dl` with `youtube-dlc`, not sure what streamlink uses, so just in case

and i have no idea how to check this problem from several sides and understand which process is responsible for this behaviour.

I apologies if this has turned out to be too much of an off-topic.

**Comment by bastimeyer** [comment, marked as answer]
Unsupported: #2936
YouTube's adaptive streams require URL signing with sophisticated logic that needs to be re-implemented from their (volatile) JS code.

Use yt-dlp or pytube instead.

**Reply by lumpsoid** [reply]
thank you for help

**Comment by lumpsoid** [comment]
found out that my problem is related to the throttling youtube system, and [only some format will be downloaded on full speed](https://github.com/yt-dlp/yt-dlp/issues/4635#issuecomment-1367164334)

**Reply by bastimeyer** [reply]
As explained, adaptive streams require URL signing, which isn't implemented in Streamlink. If no signature is provided in the HTTP request, YT will throttle the data stream.
